dist/localforage.js

```diff
@@ -1040,37 +1032,72 @@ return /******/ (function(modules) { // webpackBootstrap
 	        LocalForage.prototype.setDriver = function setDriver(drivers, callback, errorCallback) {
 	            var self = this;
 
-	            if (typeof drivers === 'string') {
+	            if (!isArray(drivers)) {
 	                drivers = [drivers];
 	            }
 
-	            this._driverSet = new Promise(function (resolve, reject) {
-	                var driverName = self._getFirstSupportedDriver(drivers);
-	                var error = new Error('No available storage method found.');
+	            var supportedDrivers = this._getSupportedDrivers(drivers);
 
-	                if (!driverName) {
-	                    self._driverSet = Promise.reject(error);
-	                    reject(error);
-	                    return;
-	                }
+	            function setDriverToConfig() {
+	                self._config.driver = self.driver();
+	            }
+
+	            function initDriver(supportedDrivers) {
+	                return function () {
+	                    var crntDriverPos = 0;
+
+	                    function driverPromiseLoop() {
+	                        while (crntDriverPos < supportedDrivers.length) {
+	                            var driverName = supportedDrivers[crntDriverPos];
+	                            crntDriverPos++;
+
+	                            self._dbInfo = null;
+	                            self._ready = null;
+
+	                            return self.getDriver(driverName).then(function (driver) {
+	                                self._extend(driver);
+	                                setDriverToConfig();
+
+	                                self._ready = self._initStorage(self._config);
+	                                return self._ready;
+	                            })['catch'](driverPromiseLoop);
+	                        }
 
+	                        setDriverToConfig();
+	                        var error = new Error('No available storage method found.');
+	                        self._driverSet = Promise.reject(error);
+	                        return self._driverSet;
+	                    }
+
+	                    return driverPromiseLoop();
+	                };
+	            }
+
+	            // There might be a driver initialization in progress
+	            // so wait for it to finish in order to avoid a possible
+	            // race condition to set _dbInfo
+	            var oldDriverSetDone = this._driverSet !== null ? this._driverSet['catch'](function () {
+	                return Promise.resolve();
+	            }) : Promise.resolve();
```

Comment:
That is one of the more impressive ternary statements I've seen in awhile! :smile: